### PR TITLE
[enhancement](ldap) optimize LDAP authentication.

### PR DESCRIPTION
**Pull request by luozenglin**
1. Support caching LDAP user information.
2. HTTP authentication supports LDAP.
3. LDAP temporary users support default user property.
4. LDAP configuration supports the `admin show config` and `admin set config` commands.

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        if (Strings.isNullOrEmpty(userName)) {
+            return null;
+        } else if (!ldapClient.doesUserExist(userName)) {
+            removeUserIfExist(fulName);
+            return null;
+        }
+        checkTimeoutCleanCache();
+
+        LdapUserInfo ldapUserInfo = new LdapUserInfo(fulName, false, "", getLdapGroupsPrivs(userName, cluster));
+        lock.writeLock().lock();
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java:119]: Would it be better to use writeLock() and writeUnlock()?

File: fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        }
+        return false;
+    }
+
+    private boolean checkParam(String fullName) {
+        return LdapConfig.ldap_authentication_enabled && !Strings.isNullOrEmpty(fullName) && !fullName.equalsIgnoreCase(
+                PaloAuth.ROOT_USER) && !fullName.equalsIgnoreCase(PaloAuth.ADMIN_USER);
+    }
+
+    private LdapUserInfo updateUserInfo(String fulName) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java:107]:
```suggestion
    private LdapUserInfo getUserInfoAndUpdateCache(String fulName) {
```

File: fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java (hunk 1 above)
Comment (luozenglin) [reply fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java]: good idea

File: fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java (hunk 2 above)
Comment (luozenglin) [reply fe/fe-core/src/main/java/org/apache/doris/ldap/LdapManager.java]: done. thank you.

File: fe/fe-core/src/main/java/org/apache/doris/ldap/LdapClient.java [hunk 3]
```diff
@@ -132,20 +132,20 @@ public static boolean doesUserExist(String userName) {
...
-    public static boolean checkPassword(String userName, String password) {
+    boolean checkPassword(String userName, String password) {
         init();
         try {
             clientInfo.getLdapTemplateNoPool().authenticate(org.springframework.ldap.query.LdapQueryBuilder.query()
                     .base(LdapConfig.ldap_user_basedn)
                     .filter(getUserFilter(LdapConfig.ldap_user_filter, userName)), password);
             return true;
-        } catch (Exception e) {
+        } catch (org.springframework.ldap.AuthenticationException e) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/ldap/LdapClient.java:142]: The range captured here is small, which will result in doris being logged in without a password.

**Review by stalary — APPROVED** [review]
LGTM

**Pull request merged**